nat-lab/tests/test_network_switch.py

```diff
@@ -332,33 +328,33 @@ async def test_mesh_network_switch_direct(
             await testing.wait_long(ping.wait_for_next_ping())
 
         await network_switcher.switch_to_secondary_network()
-
         await alpha_client.notify_network_change()
 
         await testing.wait_lengthy(
-            alpha_client.wait_for_any_derp_state(
```

Comment:
I see this doesn't work? waiting for all connection states I mean...

What if you were to write something like this:
```
async with run_async_contexts(
    [
        alpha_client.wait_for_event_on_any_derp([telio.State.Disconnected]),
        alpha_client.wait_for_event_on_any_derp([telio.State.Connecting]),
        alpha_client.wait_for_event_on_any_derp([telio.State.Connected]),
    ]
) as futures:
    # wait for futures to schedule
    await asyncio.sleep(0)
    await alpha_client.notify_network_change()
    for fut in futures:
        await fut
```

Reply:
that would work I think.. is it worth to check those intermediate events when we only care about [telio.State.Connected]? 


I have the same question for other tests: if we want to wait for a node event with Direct path is it worth to check [telio.State.Connected] Derp event and [telio.State.Connected] node event? IMO we should have those intermediate steps on a just one test case to test specifically that, on all other tests it looks redundant.

Reply:
I'd say it is important to track all the events, even if they are not really relevant to test. This way we are making sure, that correct flow is executed. But maybe let's leave this for the future.

And about direct path. It is important to check for derp first, since w/o derp, there cannot be any direct connection.

Reply:
> And about direct path. It is important to check for derp first, since w/o derp, there cannot be any direct connection.
That's the point,

Reply:
My though is to simplify tests as much as possible, to have transparent tests so that we easily understand them: adding common setup/destroy methods, cleaning test redundancy (example above), limit tests to one behaviour per test case, etc.

Reply:
I think we will do this in the future, with nat-lab framework improvements. For now this PR is not about this I guess

Reply:
correct 😄